src/hazelcore/Builtins.re

```diff
@@ -0,0 +1,183 @@
+/*
+   Built-in functions for Hazel.
+
+   To add a built-in function or constant, write the implementation in the
+   `Impls` module below and add it to the `built-ins` list.
+
+   See the existing ones for reference.
+ */
+
+open Sexplib.Std;
+
+[@deriving sexp]
+type args = list(DHExp.t);
+
+[@deriving sexp]
+type form = (
+  /* eval: */ (args, DHExp.t => EvaluatorResult.t) => EvaluatorResult.t,
+  /* elab: */ DHExp.t,
+);
+
+[@deriving sexp]
+type t = (Var.t, HTyp.t, form);
+
+module Impl = {
+  [@deriving sexp]
+  type f =
+    (
+      /* args: */ args,
+      /* evaluate: */ DHExp.t => EvaluatorResult.t,
+      /* exp: */ DHExp.t
```

Comment:
why are we passing in the initial unevaluated expression? to support Indet? if so, it isn't quite right -- the indet expression should have fully evaluated args even if it can't be further reduced, not left entirely unevaluated...

Reply:
Not sure why I did it like this originally... should be fixed now, I think.